File: src/main/kotlin/inflearn/java/ch01_string/_1_12_암호.java
package inflearn.java.ch01_string;

import java.util.Scanner;

public class _1_12_암호 {
    public static String solution(int n, String s) {
        StringBuilder answer = new StringBuilder();
        s = s.replace('#', '1');
        s = s.replace('*', '0');

        while (s.length() > 0) {
            String now = s.substring(0, 7);
            s = s.substring(7);
            char ch = (char) (Integer.parseInt(now, 2));
            answer.append(ch);
        }

        return answer.toString();
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = Integer.parseInt(sc.next());
        String s = sc.next();
        System.out.println(solution(n, s));
    }
}

File: src/main/kotlin/inflearn/kotlin/ch01/1_12 암호.kt
package inflearn.kotlin.ch01

object `1_12 암호` {
    fun solution(n: Int, s: String): String? {
        val answer = StringBuilder()

        var str = s.map { if(it == '#') '1' else '0' }.joinToString("")
        while (str.isNotEmpty()) {
            val now = str.substring(0, 7)
            str = str.substring(7)
            val ch = now.toInt(2).toChar()
            answer.append(ch)
        }
        return answer.toString()
    }

    @JvmStatic
    fun main(args: Array<String>) {
        val n = readln().toInt()
        val s = readln()
        println(solution(n, s))
    }
}